vm/actor/src/builtin/verifreg/mod.rs

```diff
@@ -0,0 +1,349 @@
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*STORAGE_MARKET_ACTOR_ADDR))?;
+        if params.deal_size < *MINIMUM_VERIFIED_SIZE {
+            return Err(ActorError::new(
+                ExitCode::ErrIllegalArgument,
+                format!(
+                    "Verified Dealsize {:} is below minimum in usedbytes",
+                    params.deal_size
+                ),
+            ));
+        }
+
+        rt.transaction(|st: &mut State, rt| {
+            let verifier_cap = st
+                .get_verified_client(rt.store(), &params.address)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("Failed to get Verifier {:?}", params.address.clone()),
+                    )
+                })?
+                .unwrap_or_else(Zero::zero);
+
+            let new_verifier_cap = verifier_cap + &params.deal_size;
+            st.put_verified_client(rt.store(), &params.address, &new_verifier_cap)
+                .map_err(|_| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!(
+                            "Failed to put verified client{:} with bytes {:}",
+                            params.address, params.deal_size
+                        ),
+                    )
+                })
+        })??;
+
+        Ok(())
+    }
+}
+
+impl ActorCode for Actor {
+    fn invoke_method<BS, RT>(
+        &self,
+        rt: &mut RT,
+        method: MethodNum,
+        params: &Serialized,
+    ) -> Result<Serialized, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        match FromPrimitive::from_u64(method) {
+            Some(Method::Constructor) => {
+                Self::constructor(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            Some(Method::AddVerifier) => {
+                Self::add_verifier(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            Some(Method::RemoveVerifier) => {
+                Self::remove_verifier(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            Some(Method::AddVerifiedClient) => {
+                Self::add_verified_client(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            Some(Method::UseBytes) => {
+                Self::use_bytes(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            Some(Method::RestoreBytes) => {
+                Self::restore_bytes(rt, params.deserialize()?)?;
+                Ok(Serialized::default())
+            }
+            _ => Err(rt.abort(ExitCode::SysErrInvalidMethod, "Invalid method".to_owned())),
```

Comment:
This only covers the `None` case, right? If so, I'd suggest changing this to `None => ...` to make this more obvious, and to not introduce bugs when e.g. a new enum variant is added